dotnet-trace/test.sh

```diff
@@ -48,14 +48,14 @@ else
    exit 1
 fi 
 
-dotnet-trace report $FILENAME topN
-if [ -f $REPORTNAME ]; then
+if dotnet-trace report $FILENAME topN | grep -q 'Missing Symbol'; then
```

Comment:
Is this saying that `Missing Symbols` is expected in the output? To check that `dotnet trace report topN` is working correctly, we should check that there are no errors about missing symbols.

Reply:
Yes, `Missing Symbol` is expected in the output in order to verify that the command is returning the methods that have been on the callstack.

Reply:
This doesn't look like the case for .NET 8.

The CI job for 8 is failing with this in the logs:

```
+ grep -q 'Missing Symbol'
+ dotnet-trace report dotnettrace.nettrace topN
+ echo 'report - FAIL'
+ rm -r dotnettrace
report - FAIL
+ rm dotnettrace.nettrace
+ rm dotnettrace.speedscope.json
+ rm dotnettrace.nettrace.etlx
+ exit 1
```

Running the topN command on my machine using .NET 8, I see output like:

```
$ dotnet trace report /home/omajid/temp/dotnet_20231002_101421.nettrace topN
Top 5 Functions (Exclusive)                                                  Inclusive           Exclusive           
1. LowLevelLifoSemaphore.WaitNative(class Microsoft.Win32.SafeHandles.Saf    57.28%              57.28%              
2. Monitor.Wait(class System.Object,int32)                                   12.94%              12.94%              
3. WaitHandle.WaitOneNoCheck(int32)                                          10.71%              10.71%              
4. WaitHandle.WaitMultiple(value class System.ReadOnlySpan`1<class System    9.04%               9.04%               
5. WaitHandle.WaitAnyMultiple(value class System.ReadOnlySpan`1<class Mic    2.43%               2.43%               
```

While .NET 7 shows:

```
$ dotnet trace report /home/omajid/temp/dotnet_20231002_112832.nettrace topN
Top 5 Functions (Exclusive)                                                  Inclusive           Exclusive           
1. Missing Symbol                                                            43.04%              42.81%              
2. WaitHandle.WaitOneNoCheck(int32)                                          10.62%              10.62%              
3. ManualResetEventSlim.Wait(int32,value class System.Threading.Cancellat    9.69%               9.69%               
4. __Canon].MoveNext()                                                       7.37%               7.33%               
5. MSBuildEvaluator.EvaluateProject(class Microsoft.TemplateEngine.Abstra    4.35%               2.21%
```

I actually think the .NET 7 output showing "Missing symbol" is a bug - it show show the symbol (method) name, not that it can't find the symbol

Reply:
Ah OK, thank you for clarifying this.

Reply:
Digging some more (eg, https://github.com/dotnet/diagnostics/issues/3900, https://github.com/dotnet/diagnostics/blob/f4e915c7e9b5869d9b3b646aadf87755061c6638/src/Tools/dotnet-trace/CommandLine/PrintReportHelper.cs#L124), it looks like this message is printed when the debug symbols are missing. The symbols can be missing due to runtime or dotnet-trace bugs or because application-specific debug symbols are simply not installed.

With .NET 7, we didn't ship managed debug symbols as part of the SDK. So that could be an issue. Or it could be a bug in runtime or dotnet-trace. With .NET 8, we are shipping debug symbols.

How about this: let's limit this test to .NET 8 and make it fail if we see any "Missing Symbol" in the output? Then we can see if/when this test fails and figure out if it was a breakage in the runtime or dotnet-trace or something else?